wallet/wallet.go

```diff
@@ -850,6 +850,125 @@ func (w *Wallet) FetchMissingCFilters(ctx context.Context, p Peer) error {
 	}
 }
 
+type MissingCFilterProgress struct {
+	Err              error
+	BlockHeightStart int32
+	BlockHeightEnd   int32
+}
+
+// FetchMissingCFiltersWithProgress records any missing compact filters for main chain
+// blocks.  A database upgrade requires all compact filters to be recorded for
+// the main chain before any more blocks may be attached, but this information
+// must be fetched at runtime after the upgrade as it is not already known at
+// the time of upgrade.  This function reports to a channel with any progress
+// that may have seen.
+func (w *Wallet) FetchMissingCFiltersWithProgress(ctx context.Context, p Peer, progress chan<- MissingCFilterProgress) {
```

Comment:
This is a lot of code duplication I'd like to avoid.  We can architect this similar to the various Rescan* methods which all call an unexported method for the common implementation.